scripts/create_text_dump_of_source_data.py

```diff
@@ -2,31 +2,56 @@
 import click
 import click_config_file
 import isb_lib.core
+import csv
 from isb_web.sqlmodel_database import SQLModelDAO
 
 excluded_keys = ["id", "bcid", "uri", "@id"]
-SEPARATOR = "###"
 
 
-def process_dictionary(dictionary: typing.Dict) -> str:
-    """Process all the keys in a dictionary and return all the values in a string, recursing through embedded dicts"""
-    line = ""
-    for key in dictionary:
+def process_dictionary(
+    base_key: typing.Optional[str],
+    field_names: set[str],
```

Comment:
This was part of the trickiness -- the records don't necessarily have consistent fields in them, so make a superset of all the fields while we iterate over them